src/lazy/encoder/binary/v1_1/flex_uint.rs

```diff
@@ -337,74 +271,25 @@ impl FlexUInt {
         // back to the general-purpose.
         cold_path! { encode_opcode_and_length_general_case => {
             output.push(opcode);
-            FlexUInt::encode_u64_general_case(output, length)
+            let _ = FlexUInt::write(output, length).unwrap();
         }}
     }
 
-    pub fn write_u64<W: Write>(output: &mut W, value: u64) -> IonResult<usize> {
+    const MAX_FLEX_UINT_SIZE_IN_BYTES: usize = 16;
```

Comment:
Encoded or decoded size?

Reply:
Encoded.

Reply:
So, the max supported encoded size is actually 18 bytes, right? 2 bytes worth of continuation bits, and then 16 bytes of data? Or are we limiting it to 16 bytes in order to simplify the implementation?